include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h

```diff
@@ -0,0 +1,1842 @@
+              else
+                weights_[fe_pair_no][i] =
+                  Number(1.) / touch_count.local_element(
+                                 level_dof_indices_fine_[fe_pair_no][i]);
+
+            level_dof_indices_fine_[fe_pair_no] +=
+              transfer.schemes[fe_pair_no].n_dofs_fine_cell;
+            weights_[fe_pair_no] +=
+              transfer.schemes[fe_pair_no].n_dofs_fine_cell;
+          });
+        }
+    }
+  };
+
+} // namespace internal
+
+
+template <int dim, typename Number>
+void
+MGTwoLevelTransfer<dim, LinearAlgebra::distributed::Vector<Number>>::prolongate(
+  const unsigned int                                to_level,
+  LinearAlgebra::distributed::Vector<Number> &      dst,
+  const LinearAlgebra::distributed::Vector<Number> &src) const
+{
+  (void)to_level;
+
+  const unsigned int vec_size = VectorizedArray<Number>::n_array_elements;
+
+  this->vec_coarse.copy_locally_owned_data_from(src);
+  this->vec_coarse.update_ghost_values();
+  this->constraint_coarse.distribute(this->vec_coarse);
+  this->vec_coarse
+    .update_ghost_values(); // note: make sure that ghost values are set
+
+  this->vec_fine = Number(0.);
+
+  for (const auto &scheme : schemes)
+    {
+      AlignedVector<VectorizedArray<Number>> evaluation_data_fine(
+        scheme.n_dofs_fine_cell);
+      AlignedVector<VectorizedArray<Number>> evaluation_data_coarse(
+        scheme.n_dofs_fine_cell);
+
+      CellProlongator<dim, VectorizedArray<Number>> cell_prolongator(
+        scheme.prolongation_matrix_1d,
+        evaluation_data_coarse.begin(),
+        evaluation_data_fine.begin());
+      CellTransfer cell_transfer(scheme.degree_fine, scheme.degree_coarse);
+
+      for (unsigned int cell = 0; cell < scheme.n_coarse_cells;
+           cell += vec_size)
+        {
+          const unsigned int n_lanes =
+            (cell + vec_size > scheme.n_coarse_cells) ?
+              (scheme.n_coarse_cells - cell) :
+              vec_size;
+
+          // read from source vector
+          {
+            const unsigned int *indices =
+              &scheme
+                 .level_dof_indices_coarse[cell * scheme.n_dofs_coarse_cell];
+            for (unsigned int v = 0; v < n_lanes; ++v)
+              {
+                for (unsigned int i = 0; i < scheme.n_dofs_coarse_cell; ++i)
+                  evaluation_data_coarse[i][v] =
+                    this->vec_coarse.local_element(indices[i]);
+                indices += scheme.n_dofs_coarse_cell;
+              }
+          }
+
+          // ---------------------------- coarse -----------------------------
+          const auto ierr = cell_transfer.run(cell_prolongator);
+          (void)ierr;
+          Assert(ierr,
+                 ExcMessage("Prolongation " +
+                            std::to_string(scheme.degree_coarse) + " -> " +
+                            std::to_string(scheme.degree_fine) +
+                            " not instantiated!"));
```

Comment:
This is the place I was referring to in my comment above about the default case `degree=-1`.